dsl/pace/dsl/dace/utils.py

```diff
@@ -120,4 +128,119 @@ def evaluate(expr):
             output_nodes={newnode.data: newnode},
             external_edges=True,
         )
-    print(f"Added {len(checks)} NaN checks")
+    logger.info(f"Added {len(checks)} NaN checks")
+
+
+def is_ref(sd: dace.sdfg.SDFG, aname: str):
+    found = False
+    for node, state in sd.all_nodes_recursive():
+        if not isinstance(state, dace.sdfg.SDFGState):
+            continue
+        if state.parent is sd:
+            if isinstance(node, dace.nodes.AccessNode) and aname == node.data:
+                found = True
+                break
+
+    return found
```

Comment:
If this is only used in this file, prepend with an underscore
```suggestion
def _is_ref(sd: dace.sdfg.SDFG, aname: str):
    """Checks if aname is referenced in any SDFGState."""
    found = False
    for node, state in sd.all_nodes_recursive():
        if not isinstance(state, dace.sdfg.SDFGState):
            continue
        if state.parent is sd:
            if isinstance(node, dace.nodes.AccessNode) and aname == node.data:
                found = True
                break

    return found
```